### ISSUE TITLE
[Question] Possible to redirect http requests through RDP?

### ISSUE THREAD

**Issue by tintanto** [fiddyschmitt/File-Tunnel#13]
Hello. I am not sure if this project is for my use case or not. I want to redirect my python https requests through the RDP server. Is this possible using this project? If that is the case, I would appreciate any tips.

**Comment by fiddyschmitt** [comment]
Hi tintanto,

Yes it should work. Perhaps your web server is an HTTP proxy, rather than a SOCKS proxy.

This works for me (I'm using mitmproxy)

```
import requests

proxies = {
    "http": "http://127.0.0.1:5000",
}

url = 'https://www.google.com/'
response = requests.get(url, proxies=proxies)
print(response.text)
```

Please let me know :)

Cheers,
Fidel

**Comment by tintanto** [comment]
@fiddyschmitt Thank you for the reply. It seems I do not comprehend how the ft project works. I thought I would actually run ft on both my PC and destination PC, where I rdp to, then connect to 127.0.0.1:5000 using my python project and that would be enough for my `requests` traffic to be redirected. I have used the two lines I have mentioned in my first post. I have also used ` {'http': 'http://127.0.0.1:5000', 'https': 'http://127.0.0.1:5000'}` and also ` {'http': 'http://127.0.0.1:5000', 'https': 'http://127.0.0.1:5000'}` with `requests.get()` That did not work. It resulted in the remote PC displaying `Error during connection to 10.1.0.48:8888. A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. [::ffff:10.1.0.48]:8888` Is there an extra step I am missing? My aim is to actually use the remote PC VPN connection. If that is not possible please let me know.

Thank you.

**Comment by fiddyschmitt** [comment]
No worries, it sounds very interesting. Let's get it working.

Is this your setup?

![image](https://github.com/user-attachments/assets/b8c03e23-39c2-41bd-a128-5de60a30d92d)

And when you do an HTTP request in your Python script, you'd like it originate out of the Remote PC, utilising the VPN connection it's running?

**Comment by tintanto** [comment]
@fiddyschmitt  Yes. That is my setup. Also, that is also what I want. I want my python script `requests` to have the IP of the remote PC active VPN service. If that is not possible with ft I can settle for being able to send the requests to the remote PC and then try to find a different solution for utilizing the VPN connection. Thank you.

**Comment by fiddyschmitt** [comment]
Yes File Tunnel doesn't act like a VPN service itself. It just tunnels TCP connections. So you'll need a Proxy server in your setup.

![image](https://github.com/user-attachments/assets/35c4848b-1146-4294-8791-05ee604a116e)

- Download [simple-proxy](https://github.com/jthomperoo/simple-proxy)
- Run `simple-proxy.exe` which listens on port 8888 by default
- Point the remote ft.exe to 8888, and your Python script should get internet access

**Comment by tintanto** [comment]
@fiddyschmitt Thank you so much for the help. I think it actually worked. Sadly, the remote PC crashed before I could do further tests. Is there any way to reduce the memory consumption of ft. The program uses over 1.6GB of RAM? If not thanks again for the help.

**Comment by fiddyschmitt** [comment]
That's great! You're welcome! I'm so glad it worked.

Shame about the crash.

I'll look to see if there's a way to reduce memory consumption in the coming weeks.

**Comment by fiddyschmitt** [comment]
This is now supported in v3.4.0.

Use -D (dynamic/SOCKS forwarding, like ssh -D):

On the machine running Python (creates a SOCKS5 proxy on 127.0.0.1:5000; the RDP server dials out):
`ft.exe -D 5000 --write "C:\Temp\1.dat" --read "C:\Temp\2.dat"`

On the RDP server:
`ft.exe --read "\\tsclient\c\Temp\1.dat" --write "\\tsclient\c\Temp\2.dat"`

Your Python was correct - just add `rdns=True` so DNS is also resolved at the RDP server:

```
socks.set_default_proxy(socks.SOCKS5, "127.0.0.1", 5000, rdns=True)
socket.socket = socks.socksocket
```
